### ISSUE TITLE
Button component sizing issue

### ISSUE THREAD

**Issue by kufelkamil** [looker-open-source/components#2921]
Starting from version 2.8.7 `Button` component behaves unexpectedly, it shrinks too much when used inside `Space` along with other components. Maybe the bug was introduced in 2.8.6 as a part of adding ripple effect (according to the changelog), but I cannot choose 2.8.6 on codesandbox. 

https://codesandbox.io/s/beautiful-hill-oxs3f?file=/src/index.tsx

<img width="1380" alt="obraz" src="https://user-images.githubusercontent.com/9267248/141839367-95eb6ad8-56fc-48cc-abd4-66192f5900a0.png">

<img width="1388" alt="obraz" src="https://user-images.githubusercontent.com/9267248/141839758-7e04493e-9909-4a25-8e9c-dc25b8d6a1c6.png">

Workaround is to wrap `Button` in `div`, like 
```
<Space>
    <div><Button iconBefore={<Pivot />}>Add something</Button></div>
    ...
</Space>
```

**Comment by kufelkamil** [comment]
Hi, in version 2.8.13 issue still exists, any chance to fix it?

**Comment by mdodgelooker** [comment]
Hi @kufelkamil thank you for raising this issue, and apologies for the delay. We're working on a fix now.

**Comment by mdodgelooker** [comment]
Hi @kufelkamil this issue is fixed in v2.8.14: https://codesandbox.io/s/admiring-smoke-ge5kw
